### PR TITLE
Optimise download all query

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Optimise data download query and improve maintainability.

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [x] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: backend-api/app/repository/sql/download.sql [hunk 1]
```diff
@@ -1,358 +1,195 @@
...
-            created DESC
-    ),
+-- Optimised version with improved performance
+WITH
+-- Single CTE for all slug deduplication (family and document)
+recent_slugs AS (
+    SELECT DISTINCT ON (
+        COALESCE(family_import_id::TEXT, '')
+        || '|'
+        || COALESCE(family_document_import_id::TEXT, '')
```
Comment (henry-cpr) [inline backend-api/app/repository/sql/download.sql:8]: Nonblocking suggestion: COACLESCE can be a bit weird with indexes. Did you run explain analyze (when not under massive time constraint i guess!) to work out if performance is all good?

**Review by henry-cpr — APPROVED** [review]
I think in an ideal world we'd have a old_query() == new_query() set of tests and check edge cases for documents with NULL slugs and duplicate slugs. But it looks good to me and is a lovely reduction in code size & improvement in readability :)
